diff --git a/src/posix/platform/CMakeLists.txt b/src/posix/platform/CMakeLists.txt
index 8915f7d4a..4737f1039 100644
--- a/src/posix/platform/CMakeLists.txt
+++ b/src/posix/platform/CMakeLists.txt
@@ -173,29 +173,28 @@ target_include_directories(openthread-posix PRIVATE
 )

 if(OT_POSIX_CONFIG_RCP_BUS MATCHES "CPC")
-    set(CPCD_SOURCE_DIR "" CACHE STRING "Location of cpc daemon project")
-    CHECK_INCLUDE_FILE(sl_cpc.h CPC_HEADER_FOUND)
-    if (NOT CPC_HEADER_FOUND)
-        message(STATUS "CPC is not installed.")
+    if (NOT TARGET cpc)
+        message(WARNING "CPC target is not available.")
+
         if (NOT CPCD_SOURCE_DIR)
             message(FATAL_ERROR "Could not locate CPC daemon sources.\n"
                                 "Please define the CMake variable 'CPCD_SOURCE_DIR'.\n"
                                 "'CPCD_SOURCE_DIR' is an absolute path to the CPC Daemon sources")
         endif()

-        message(STATUS  "Building cpc library from source...\n"
-                        "    CPCD_SOURCE_DIR = ${CPCD_SOURCE_DIR}")
         if (NOT IS_ABSOLUTE ${CPCD_SOURCE_DIR})
             message(FATAL_ERROR "'CPCD_SOURCE_DIR' must be an absolute path.\nCPCD_SOURCE_DIR=\"${CPCD_SOURCE_DIR}\"")
+        else()
+            message(WARNING "Building cpc library from source...\n"
+                            "    CPCD_SOURCE_DIR = ${CPCD_SOURCE_DIR}")
         endif()

         # Add cpcd to the build tree
         add_subdirectory(${CPCD_SOURCE_DIR} ${CPCD_SOURCE_DIR}/build)
-        target_include_directories(openthread-posix PRIVATE
-            ${CPCD_SOURCE_DIR}/lib
-        )
     endif()

+    get_target_property(CPCD_SOURCE_DIR cpc SOURCE_DIR)
+    target_include_directories(openthread-posix PRIVATE ${CPCD_SOURCE_DIR}/lib)
     target_sources(openthread-posix PRIVATE cpc_interface.cpp)
     target_link_libraries(openthread-posix PRIVATE cpc)

